src/ring_theory/fractional_ideal.lean

```diff
@@ -819,12 +806,17 @@ begin
   apply (le_div_iff_of_nonzero hI).mpr _,
   intros y hy x hx,
   rw mul_comm,
-  exact mul_mem_mul hx hy
+  exact mul_mem_mul hx hy,
 end
 
-theorem mul_inv_cancel_iff {I : fractional_ideal g} :
-  I * I⁻¹ = 1 ↔ ∃ J, I * J = 1 :=
-⟨λ h, ⟨I⁻¹, h⟩, λ ⟨J, hJ⟩, by rwa [←right_inverse_eq I J hJ]⟩
+theorem self_div_cancel_iff {I : fractional_ideal g} :
+  I * (1 / I) = 1 ↔ ∃ J, I * J = 1 := --sorry
+begin
+  split, all_goals {intro h},
+  { use 1 / I, assumption },
+  { rcases h with ⟨J, hJ⟩,
+    rwa ← prod_one_self_div_eq I J hJ },
+end
```

Comment:
Does the term mode proof (changing `right_inverse_eq` to `prod_one_self_div_eq` and `I⁻¹` to `1 / I`) not work?

Reply:
Oh yes, it does work! I have changed it.